### PR TITLE
Fix hack record

### PR DESCRIPTION
**Pull request by zhoujia6139**
## Security Checklist

- [ ] 1. Re-Entrancy
- [ ] 2. Arithmetic Over/Under Flows
- [ ] 3. Unexpected Ether
- [ ] 4. Delegatecall
- [ ] 5. Default Visibilities
- [ ] 6. Entropy Illusion
- [ ] 7. External Contract Referencing
- [ ] 8. Short Address/Parameter Attack (off chain)
- [ ] 9. Unchecked CALL Return Values
- [ ] 10. Race Conditions / Front Running
- [ ] 11. Denial Of Service (DOS)
- [ ] 12. Block Timestamp Manipulation
- [ ] 13. Constructors with Care
- [ ] 14. Uninitialized Storage Pointers
- [ ] 15. Floating Points and Precision
- [ ] 16. Tx.Origin Authentication
- [ ] 17. Address.isContract Re-Entrancy via Constructor

### ⚠️ NOTES ⚠️

Make sure to think about each of these exploits in this PR.

### CODE REVIEW

File: contracts/misc/AutoCompoundApe.sol [hunk 1]
```diff
@@ -29,6 +30,7 @@ contract AutoCompoundApe is
     ApeCoinStaking public immutable apeStaking;
     IERC20 public immutable apeCoin;
     uint256 public bufferBalance;
+    uint256 public bufferStakingBalance;
```
Comment (i5possible) [inline contracts/misc/AutoCompoundApe.sol:33]:
```suggestion
    uint256 public stakingBalance;
```
It's not a buffer that will be cleared. So maybe we use stakingaBalance or stakingAmount directly?

File: contracts/protocol/libraries/logic/SupplyLogic.sol [hunk 2]
```diff
@@ -69,6 +70,81 @@ library SupplyLogic {
...
+    }
+
+    function executeWithdrawUserPosition(
+        mapping(address => DataTypes.ReserveData) storage reservesData,
+        address user,
+        address asset
+    ) external {
+        DataTypes.ReserveData storage currentReserve = reservesData[asset];
+        address xTokenAddress = currentReserve.xTokenAddress;
+        uint256 liquidityIndex = currentReserve.liquidityIndex;
```
Comment (GopherJ) [inline contracts/protocol/libraries/logic/SupplyLogic.sol:141]: we didn't updateState here

File: contracts/protocol/libraries/logic/SupplyLogic.sol [hunk 3]
```diff
@@ -69,6 +70,81 @@ library SupplyLogic {
...
+        uint256 pBalance = IPToken(xTokenAddress).balanceOf(user);
+        if (pBalance > 0) {
+            IPToken(xTokenAddress).burn(user, user, pBalance, liquidityIndex);
```
Comment (GopherJ) [inline contracts/protocol/libraries/logic/SupplyLogic.sol:144]: we may want to just burn ptoken without sending assets to user, so shall we define receiver in the parameters

File: contracts/misc/CApe.sol [hunk 4]
```diff
@@ -348,6 +349,7 @@ abstract contract CApe is ContextUpgradeable, ICApe, PausableUpgradeable {
         address _recipient,
         uint256 _sharesAmount
     ) internal whenNotPaused {
+        revert("paused");
```
Comment (GopherJ) [inline contracts/misc/CApe.sol:352]: we still need to deposit APE as cAPE then supply to lending right?

File: contracts/protocol/libraries/logic/ValidationLogic.sol [hunk 5]
```diff
@@ -388,20 +388,20 @@ library ValidationLogic {
             DataTypes.AssetType assetType
         ) = reserveCache.reserveConfiguration.getFlags();
         require(isActive, Errors.RESERVE_INACTIVE);
-        require(!isPaused, Errors.RESERVE_PAUSED);
+        // require(!isPaused, Errors.RESERVE_PAUSED);
```
Comment (GopherJ) [inline contracts/protocol/libraries/logic/ValidationLogic.sol:391]: user will be able to repay, is it expected?

File: contracts/misc/AutoCompoundApe.sol (hunk 1 above)
Comment (zhoujia6139) [reply contracts/misc/AutoCompoundApe.sol]: ok

File: contracts/protocol/libraries/logic/SupplyLogic.sol (hunk 2 above)
Comment (zhoujia6139) [reply contracts/protocol/libraries/logic/SupplyLogic.sol:141]: yeah.

**Pull request closed without merging**